test/lib/jdk/test/lib/classloader/ClassUnloadCommon.java

```diff
@@ -67,6 +70,40 @@ public static void triggerUnloading() {
         wb.fullGC();  // will do class unloading
     }
 
+    /**
+     * Calls triggerUnloading() in a retry loop for 2 seconds until WhiteBox.isClassAlive
+     * determines that no classes named in classNames are alive.
+     *
+     * This variant of triggerUnloading() accommodates the inherent raciness
+     * of class unloading. For example, it's possible for a JIT compilation to hold
+     * strong roots to types (e.g. in virtual call or instanceof profiles) that
+     * are not released or converted to weak roots until the compilation completes.
+     *
+     * @return the set of classes that have not been unloaded after exiting the retry loop
+     */
+    public static Set<String> triggerUnloading(List<String> classNames) {
+        WhiteBox wb = WhiteBox.getWhiteBox();
+        Set<String> aliveClasses = new HashSet<>(classNames);
+        int attempt = 0;
+        while (!aliveClasses.isEmpty() && attempt < 20) {
+            ClassUnloadCommon.triggerUnloading();
+            for (String className : classNames) {
+                if (aliveClasses.contains(className)) {
+                    if (wb.isClassAlive(className)) {
+                        try {
+                            Thread.sleep(100);
+                        } catch (InterruptedException ex) {
+                        }
+                    } else {
+                        aliveClasses.remove(className);
+                    }
+                }
+            }
```

Comment:
Could we use a `Iterator` to iterate the set and remove the element? It seems we only need to iterate the alive classes instead of all the classes everytime.

Reply:
It's not worth the extra complexity of handling concurrent modification. The `classNames` list is only expected to have vey few elements.